firmware/mcal/stm32f767/os/mutex.h

```diff
@@ -0,0 +1,34 @@
+/// @author Matteo Tullo
+/// @date 2024-02-24
+
+#pragma once
+
+#include <cstdint>
+
+#include "shared/util/os.h"
+#include "shared/os/mutex.h"
+#include "cmsis_os2.h"
+
+namespace mcal::os {
+
+// TODO: Add comments and handle errors more robustly
+class Mutex final : public shared::os::Mutex  {
+private:
+    osMutexId_t* mutex_id;
```

Comment:
wrong naming style

Reply:
also, variable should not be a pointer to an ID. The ID is already a pointer so it is a lightweight object